lib/vtls/nss.c

```diff
@@ -1512,6 +1513,80 @@ static CURLcode nss_load_ca_certificates(struct connectdata *conn,
   return CURLE_OK;
 }
 
+static CURLcode
+set_ssl_version_min_max(SSLVersionRange *sslver, struct Curl_easy *data,
+                        struct connectdata *conn)
+{
+  long ssl_version = SSL_CONN_CONFIG(version);
+  long ssl_version_max = retrieve_ssl_version_max(ssl_version,
+                                                 SSL_CONN_CONFIG(version_max));
+  (void) data;
+
+  switch(ssl_version) {
+    case CURL_SSLVERSION_TLSv1_0:
+      sslver->min = SSL_LIBRARY_VERSION_TLS_1_0;
+      break;
+    case CURL_SSLVERSION_TLSv1_1:
+#ifdef SSL_LIBRARY_VERSION_TLS_1_1
+      sslver->min = SSL_LIBRARY_VERSION_TLS_1_1;
+      break;
+#else
+      failf(data, "TLS1.1 minor version cannot be set");
+      return CURLE_SSL_CONNECT_ERROR;
+#endif
+    case CURL_SSLVERSION_TLSv1_2:
+#ifdef SSL_LIBRARY_VERSION_TLS_1_2
+      sslver->min = SSL_LIBRARY_VERSION_TLS_1_2;
+      break;
+#else
+      failf(data, "TLS1.2 minor version cannot be set");
+      return CURLE_SSL_CONNECT_ERROR;
+#endif
+    case CURL_SSLVERSION_TLSv1_3:
+#ifdef SSL_LIBRARY_VERSION_TLS_1_3
+      sslver->min = SSL_LIBRARY_VERSION_TLS_1_3;
+      break;
+#else
+      failf(data, "TLS1.3 minor version cannot be set");
+      return CURLE_SSL_CONNECT_ERROR;
+#endif
+  }
+
+  switch(ssl_version_max) {
+    case CURL_SSLVERSION_MAX_TLSv1_0:
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_0;
+      break;
+    case CURL_SSLVERSION_MAX_TLSv1_1:
+#ifdef SSL_LIBRARY_VERSION_TLS_1_1
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_1;
+#else
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_0;
+#endif
+      break;
+    case CURL_SSLVERSION_MAX_TLSv1_2:
+#ifdef SSL_LIBRARY_VERSION_TLS_1_2
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_2;
+#elif defined(SSL_LIBRARY_VERSION_TLS_1_1)
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_1;
+#else
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_0;
+#endif
+      break;
+    case CURL_SSLVERSION_MAX_TLSv1_3:
+#ifdef SSL_LIBRARY_VERSION_TLS_1_3
+        sslver->max = SSL_LIBRARY_VERSION_TLS_1_3;
+#elif defined(SSL_LIBRARY_VERSION_TLS_1_2)
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_2;
+#elif defined(SSL_LIBRARY_VERSION_TLS_1_1)
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_1;
+#else
+      sslver->max = SSL_LIBRARY_VERSION_TLS_1_0;
+#endif
+      break;
+  }
+  return CURLE_OK;
+}
+
```

Comment:
CURL_SSLVERSION_MAX_DEFAULT is now as separated value and for NSS it uses maximum version from SSL_VersionRangeGetDefault. For other libraries it is CURL_SSLVERSION_MAX_TLSv1.2.